subprojects/base-services/src/main/java/org/gradle/internal/operations/DefaultBuildOperationQueue.java

```diff
@@ -186,10 +220,10 @@ private void runOperation(T operation) {
             }
         }
 
-        private void addFailure(Throwable failure) {
+        private void operationsFinished(int count) {
             lock.lock();
             try {
-                failures.add(failure);
+                pendingCount = pendingCount - count;
```

Comment:
need to signal on the `workDone` condition as this is what the waiting thread is blocked on. Perhaps rename the condition to make this a bit clearer.